docs/src/dev/provider/index.asciidoc

```diff
@@ -1147,25 +1147,64 @@ one key value pair present (since only one `Traversal` is being submitted, there
 single alias).
 |=========================================================
 
-=== Authentication
+=== Authentication and authorization
```

Comment:
nit: Capital "A" in "Authorization" please since it's a title. 
nit: There's a bit more formatting to do in the text like enclosing class names in backticks.

I think it would be worth adding some note here to providers to say that while Gremlin Server supports this authorization feature it is not a feature that TinkerPop requires of graph providers as part of the agreement between client and server. Graph providers may choose to implement their own methods for authorization in the manner they see fit.  I would say a similar "IMPORTANT" callout box should probably be added to the reference documentation to alert users to this notion. Finally, as you draw closer to a final body of work, this is a neat new feature that should have upgrade documentation. (and perhaps more user facing documentation?))

Reply:
Fair - I must admit that this was the easiest way to start with the documentation. Adding to the ref docs implies reorganizing the gremlin server chapter because the security section is already way too big.

Reply:
As expansive as our documentation is, we still continue to get feedback that it is confusing, incomplete, etc. Perhaps reorganization wouldn't be a really bad thing except that it means "more work". Let's see what your next iteration of this PR looks like and we can discuss further then.